### PR TITLE
Added Clerk README

### PR DESCRIPTION
**Pull request by 0xKrishna**
# Description

Added Clerk Module README

# Changes

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [ ] Changes only for a subset of nodes

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Nodes audience

In case this PR includes changes that must be applied only to a subset of nodes, please specify how you handled it (e.g. by adding a flag with a default value...)

# Checklist

- [ ] I have added at least 2 reviewer or the whole pos-v1 team
- [x] I have added sufficient documentation in code
- [ ] I will be resolving comments - if any - by pushing each fix in a separate commit and linking the commit hash in the comment reply

# Cross repository changes

- [ ] This PR requires changes to bor 
  - In case link the PR here:
- [ ] This PR requires changes to matic-cli
  - In case link the PR here:

## Testing

- [ ] I have added unit tests
- [ ] I have added tests to CI
- [ ] I have tested this code manually on local environment
- [ ] I have tested this code manually on remote devnet using express-cli
- [ ] I have tested this code manually on mumbai
- [ ] I have created new e2e tests into express-cli

### Manual tests

Please complete this section with the steps you performed if you ran manual tests for this functionality, otherwise delete it

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it

### CODE REVIEW

**Review by marcello33 — APPROVED** [review]

File: clerk/README.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+* [Preliminary terminology](#preliminary-terminology)
+* [Overview](#overview)
+* [State-Sync Mechanism](#state-sync-mechanism)
+* [How does it work](#how-does-it-work)
+* [How to add an event](#how-to-add-an-event)
+* [Query commands](#query-commands)
+
+## Preliminary terminology
+
+* A `StateSender` is a contract deployed on L1 (Ethereum) responsible for emiiting state-sync events.
```
Comment (Raneet10) [inline clerk/README.md:14]: nit: `emitting` instead of `emiiting`

Comment (0xKrishna) [reply clerk/README.md:14]: Wow, Sharp eyes boy.

File: clerk/README.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+* A `StateReceiver` is a contract deployed on L2 (Bor) responsible for receiving state-sync events.
+* A `EventRecord` is a record of the state-sync event stored in the heimdall state.
+
+## Overview
+
+Clerk manages generic event records from Ethereum blockchain related to state-sync events, These are specically designed events that are emitted by the StateSender contract on the L1 chain to notify the L2 nodes (Bor in case of PoS) about the state changes in the L1. Once the events are processed by the bridge, Clerk listens to these events and stores them in the database for further processing.
+
+## State-Sync Mechanism
+
+It's a mechansism for state-management between Ethereum and Bor chain, The events generated are called state-sync events. This is a way to move data from the L1 chain to L2 chain.
```
Comment (Raneet10) [inline clerk/README.md:24]: nit: `mechanism` rather than `mechansism` 😅

File: clerk/README.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+[Handler](handler.go) for this transaction validates for multiple conditions including `TxHash` and `LogIndex` to ensure that the event exists on L1 and the data is not tampered with, It throws `Older invalid tx found` error if the event is already processed.
+
+Once the event is validated by the Handler, It will go to `SideHandleMsgEventRecord` in each validator node and after verifying the event, The validators will vote with either a `YES` return an error for a failed verification.
+
+Only when there is a majority of `YES` votes, The event will be processed by `PostHandleMsgEventRecord` which will persist the event in the state via keeper.
+
+## How to add an event
+
+A validator can leverage the CLI to add en event to the state in case it's missing and not processed by the bridge, The CLI command is :
```
Comment (Raneet10) [inline clerk/README.md:75]: nit: `add an event`

**Review by Raneet10 — APPROVED** [review]
LGTM!

**Pull request merged**